### PR TITLE
Fix: Corrected the grid layout when screen is rotated.

### LINKED ISSUE #1206: Bug in Grid Layout ( when rotating phone)
**Actual Behaviour**
Open the app in portrait mode and open speakers fragment. It will look something like this.

screenshot_20170302-160623

Now close the app and reopen it in landscape mode and open the speakers fragment. It will look something like this.

screenshot_20170302-160647

Now rotate the screen. The layout changes to this, which is weird. It should be like first picture.

screenshot_20170302-160659

This bug also exists in bookmarks fragment, location fragment, track session activity.

screenshot_20170302-163248

**Expected Behaviour**

The layout should look like first photo after rotation 

**Would you like to work on the issue?**
yes

### PR DESCRIPTION
**Pull request by chiragw15**
Fixes #1206 
Changes: Now when rotated, the layout comes back to its normal form.

### CODE REVIEW

**Comment by harshitagupta30** [comment]
Can you please remove Fixes from commit message?

File: android/app/src/main/java/org/fossasia/openevent/activities/SpeakerDetailsActivity.java [hunk 1]
```diff
@@ -158,10 +163,17 @@ public void onServiceDisconnected(ComponentName name) {
         biography.setText(Html.fromHtml(selectedSpeaker.getShortBiography()));
         biography.setMovementMethod(LinkMovementMethod.getInstance());
 
+        //setting the grid layout to cut-off white space in tablet view
+        DisplayMetrics displayMetrics = this.getResources().getDisplayMetrics();
+        float width = displayMetrics.widthPixels / displayMetrics.density;
+        int spanCount = (int) (width/250.00);
+
+        sessionRecyclerView.setHasFixedSize(true);
+        gridLayoutManager = new GridLayoutManager(this,spanCount);
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/SpeakerDetailsActivity.java]: reformat the code.

File: android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java [hunk 2]
```diff
@@ -75,7 +77,7 @@ protected void onResume() {
         int spanCount = (int) (width/250.00);
 
         sessionRecyclerView.setHasFixedSize(true);
-        final GridLayoutManager gridLayoutManager = new GridLayoutManager(this,spanCount);
+        gridLayoutManager = new GridLayoutManager(this,spanCount);
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: `gridLayoutManager = new GridLayoutManager(this,spanCount);`
Reformat this line in all the files.

**Comment by chiragw15** [comment]
@harshitagupta30 The commit message doesn't contain "Fixes". It is according to the guidelines. The message you are talking about is title of the PR.

**Comment by arp95** [comment]
@harshitagupta30 
i think the guidelines say you can have "fixes" as a footer. So hes fine here !

File: android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java (hunk 2 above)
Comment (chiragw15) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: @harshitagupta30 Sorry, I didn't get you here. Do you want me to change this line 
`gridLayoutManager = new GridLayoutManager(this,spanCount);`  to 
`gridLayoutManager = new GridLayoutManager(this, spanCount);`

Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: ya!

**Comment by harshitagupta30** [comment]
@chiragw15 Sorry! my bad! :+1: 
@arp95 I was talking about the commit message the title not about the footer. thanks! :+1:

File: android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java [hunk 3]
```diff
@@ -29,17 +30,17 @@
...
+    @BindView(R.id.txt_no_sessions)
+    TextView noSessionsView;
+    @BindView(R.id.toolbar_locations)
+    Toolbar toolbar;
     private SessionsListAdapter sessionsListAdapter;
-
+    private GridLayoutManager gridLayoutManager;
     private List<Session> mSessions;
-    private static final int locationWiseSessionList = 1;
-
```
Comment (mananwason) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: dude you need to stop refactoring everything. It seems that this is an attempt to increase the number of lines of code modified. We will just close PR's where we see this kind of behaviour in the future.

Comment (chiragw15) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: Sorry @mananwason , I was asked to reformat the code as per review by @harshitagupta30 . Correcting it.

Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: Sorry @chiragw15 but I just asked to reformat the code which you've added not the entire file :) :+1:

**Comment by chiragw15** [comment]
@mananwason @harshitagupta30 Sorry, It was a misunderstanding. Corrected it.

**Pull request merged**